src/reflect/scala/reflect/internal/StdNames.scala

```diff
@@ -312,6 +314,11 @@ trait StdNames {
     override type NameType = TermName
 
     protected implicit def createNameType(name: String): TermName = newTermNameCached(name)
+    // create a name with leading dollar, avoiding the appearance that interpolation of a value (possibly named "value") was intended
+    private class NameContext(s: String) {
+      def n(args: Any*): TermName = { require(args.isEmpty) ; createNameType("$" + s) }
+    }
+    private def StringContext(parts: String*) = { require(parts.size == 1) ; new NameContext(parts.head) }
```

Comment:
👍 haven't seen this before

Reply:
The puzzler guy didn't like my one-liner StringContext puzzler. :(  It was too improbable.

Reply:
For some reason, we aren't allowed to have fun things.